docs: "What's new in Svelte" October newsletter (#1559)
sveltejs/svelte.dev: apps/svelte.dev/content/blog/2025-10-01-whats-new-in-svelte-october-2025.md

---
title: "What's new in Svelte: October 2025"
description: 'Improved Remote Functions, sv create from playgrounds and experimental async SSR'
author: Dani Sandoval
authorURL: https://dreamindani.com
---

> There were lots of improvements to remote functions this month - including new batching tools and improved performance via lazy discovery. For more info, check out the Docs and PR links in each bullet. Async SSR also has a new config flag for those who would like to try it out while it's still experimental.

Reviewer: ```suggestion
There were lots of improvements to remote functions this month - including new batching tools and improved performance via lazy discovery. For more info, check out the Docs and PR links in each bullet. Async SSR is also available for those who would like to try it out while it's still experimental. It's included when the `config.experimental.async` setting is enabled.
```

We've got a hefty showcase this month too, so let's dive in!

## What's new in Svelte, SvelteKit and Svelte CLI (sv)

- The `event` property is now included on popstate, link and form navigation (**kit@2.40.0**, [#14307](https://github.com/sveltejs/kit/pull/14307))
- The `%sveltekit.version%` placeholder can now be used in `app.html` to easily include the current SvelteKit version in the output (**kit@2.41.0**, [Docs](https://svelte.dev/docs/kit/project-structure#Project-files-src), [#12132](https://github.com/sveltejs/kit/pull/12132))

> - You can now create projects from the from the Svelte playground with `npx sv create --from-playground <url>` (**sv@0.9.5**, [#662](https://github.com/sveltejs/cli/pull/662))

Reviewer: ```suggestion
- You can now create projects from the Svelte playground with `npx sv create --from-playground <url>` (**sv@0.9.5**, [#662](https://github.com/sveltejs/cli/pull/662))
```

- Experimental async SSR is available in Svelte v5.39.3 and SvelteKit v2.43.0 or higher. You can opt-in via the `experimental.async` option. When this is enabled, it's possible to use `await` anywhere in your app, without wrapping it in a boundary with a pending snippet. You can find more information in the SvelteKit PR ([#14447](https://github.com/sveltejs/kit/pull/14447), Svelte PR is [#16748](https://github.com/sveltejs/svelte/pull/16748))

### Remote Functions

> - The remote function `query.set()` can now be called on the server and `query.refresh()` promises will now be automatically resolved on the server (**kit@2.37.0**, [Docs](https://svelte.dev/docs/kit/remote-functions), [#14304](https://github.com/sveltejs/kit/pull/14304), [#14332](https://github.com/sveltejs/kit/pull/14332))

Reviewer 2: This part seems a bit unclear. Based on this information, I could see people thinking that they could do something like this:
```js
import { query } from '$app/server';
query.set(...);
```

> - `query.batch` has been added to remote functions - letting you batch requests that happen within the same macrotask (**kit@2.38.0**, [Docs](https://svelte.dev/docs/kit/remote-functions#query.batch), [#14272](https://github.com/sveltejs/kit/pull/14272))

Reviewer 2: ```suggestion
- The new `query.batch` remote function lets you batch requests that happen within the same macrotask (**kit@2.38.0**, [Docs](https://svelte.dev/docs/kit/remote-functions#query.batch), [#14272](https://github.com/sveltejs/kit/pull/14272))
```

- Lazy discovery of remote functions improves detection and tree shaking for functions defined in `node_modules` (**kit@2.39.0**, [#14293](https://github.com/sveltejs/kit/pull/14293))

> - Remote form functions have been enhanced with schema support, `input` and `issues` properties (**kit@2.42.0**, [Docs](https://svelte.dev/docs/kit/remote-functions#form), [#14383](https://github.com/sveltejs/kit/pull/14383))

Reviewer 2: ```suggestion
- The `form` remote function has been enhanced with schema support, `input` and `issues` properties (**kit@2.42.0**, [Docs](https://svelte.dev/docs/kit/remote-functions#form), [#14383](https://github.com/sveltejs/kit/pull/14383))
```
Not sure if this is better or not, wdyt?

- [Svelte Radio](https://share.transistor.fm/s/a5f8c4c6) is back! Feat. Kevin, Brittney and Antony

_This Week in Svelte_

- [Ep. 115](https://www.youtube.com/watch?v=PHk7YFAFvfg) — Changelog review
- [Ep. 116](https://www.youtube.com/watch?v=sTepLQwJIVo) — e18e.dev and NPM's supply chain attack
- [Ep. 117](https://www.youtube.com/watch?v=E0xQXa4qQGY) — Formisch
- [Ep. 118](https://www.youtube.com/watch?v=eBXjXfUiuiA) — Changelog review

_Svelte London - September 2025_

You can [watch the full video on YouTube](https://www.youtube.com/watch?v=odCWAAaTNSQ) or skip to the specific talks below:

- [Svelte on Cloudflare: Tips and guide](https://www.youtube.com/live/odCWAAaTNSQ?si=2Eej24zu2ZXUEljs&t=412) by Mark Ridge and Connor Adams
- [Ultra-low Latency in SvelteKit](https://www.youtube.com/live/odCWAAaTNSQ?si=4Eyh2SUwA26waECz&t=2563) by Scott Phillips

_To Read_

- [Sharing some custom components that handle async state with less boilerplate in Svelte 5](https://www.reddit.com/r/sveltejs/comments/1niyflq/sharing_some_custom_components_that_handle_async/) by u/geokidev
- [Rendering emails with Svelte](https://github.com/GauBen/svelte-emails) by Gautier Ben Aïm

> 

Reviewer 3: ```suggestion
- [Integrate Keystatic CMS with SvelteKit to render Markdoc Content with Interactive Svelte Components](https://samuelplumppu.se/blog/keystatic-sveltekit-markdoc) by Samuel Plumppu

```

_Dev Tools and Vite Plugins_

- [wuchale](https://github.com/wuchalejs/wuchale) is a compile-time internationalization (i18n) toolkit that requires zero code changes
- [SV Floating Console](https://www.npmjs.com/package/sv-console) is a floating console for Svelte applications that only appears in development mode
- [magicons](https://github.com/propolies/magicons) is a fast, typesafe Icon wrapper that fixes bundling issues with large barrel exports in icon libraries
- [Svelte-pops](https://svelte-pops.vercel.app/docs/overview) is a performant and easy to use popover manager for creating dialog and tooltips
- [vite-plugin-svelte-inline-component](https://github.com/hanielu/vite-plugin-svelte-inline-component) lets you write tiny Svelte components straight inside your JavaScript / TypeScript tests using tagged‑template literals
- [vite-plugin-sveltekit-decorators](https://github.com/KiraPC/vite-plugin-sveltekit-decorators) is a powerful Vite plugin that automatically decorates SvelteKit functions with customizable wrappers for logging, analytics, error handling, and more

> 

Reviewer 3: ```suggestion
[keystatic-sveltekit](https://github.com/Greenheart/keystatic-sveltekit) is a convenient way to integrate the open source Git-based Keystatic CMS with SvelteKit, featuring Markdoc content with interactive Svelte components and a modern editing UX for your entire team.

```
